blocks/feed-newsroom/feed-newsroom.js

```diff
@@ -191,13 +195,35 @@ export default async function decorate(block) {
 </form>`;
   const uniqYears = Array.from(new Set(results.map((x) => { const itsDate = getFormattedDate(new Date(parseInt(x[blockCfg.sort.trim().toLowerCase()], 10))).split(', '); return parseInt(itsDate[itsDate.length - 1], 10); })));
   // eslint-disable-next-line
-  const yroptions = uniqYears.reduce((accum, current) => { accum += "<option value='"+current+"'>" + current + "</option>"; return accum;},"");
-  filterDiv.querySelector('select').innerHTML = filterDiv.querySelector('select').innerHTML + yroptions;
+  const yroptions = uniqYears.reduce((accum, current) => { accum += "<option value='" + current + "'>" + current + "</option>"; return accum; }, "");
+  filterDiv.querySelector('#news_year').innerHTML = filterDiv.querySelector('#news_year').innerHTML + yroptions;
+  const categoryDetails = await fetchTagsOrCategories('', 'categories', 'newsroom', locale);
+  // eslint-disable-next-line
+  const categoryOptions = categoryDetails.reduce((accum, current) => { accum += "<option value='" + current.id + "'>" + current.name + "</option>"; return accum; }, "");
+  filterDiv.querySelector('#news_category').innerHTML = filterDiv.querySelector('#news_category').innerHTML + categoryOptions;
 
   filterDiv.querySelector('form .filter-nav button').addEventListener('click', () => {
-    const searchYear = Number(filterDiv.querySelector('form .filter-nav select').value);
-    searchResults = results.filter((x) => { const itsDate = getFormattedDate(new Date(parseInt(x[blockCfg.sort.trim().toLowerCase()], 10))).split(', '); return (parseInt(itsDate[itsDate.length - 1], 10) === searchYear); });
-    loadYearResults(block, blockType, searchResults, blockCfg, locale);
+    const searchYear = Number(filterDiv.querySelector('form .filter-nav #news_year').value);
+    const searchCategory = filterDiv.querySelector('form .filter-nav #news_category').value;
+    let filteredResults = window.jslinq(results);
+    if (searchYear) {
+      filteredResults = filteredResults.where((el) => {
+        const itsDate = getFormattedDate(new Date(parseInt(el[blockCfg.sort.trim().toLowerCase()], 10))).split(', ');
+        return (parseInt(itsDate[itsDate.length - 1], 10) === searchYear);
+      });
+    }
+    if (searchCategory) {
+      filteredResults = filteredResults.where((el) => el.category === searchCategory);
+    } else if (searchYear) {
+      const option = filterDiv.querySelector('#news_category').options;
+      for (let i = 0; i < option.length; i += 1) {
+        if (option[i].text === 'News') {
+          option[i].selected = true;
+          break;
+        }
+      }
```

Comment:
What's the purpose of this `else` ? Is it to ensure that "News" category is selected automatically if no category was selected? If yes, should the "News" category also be used as a filter. Asking this as I don't see that happening currently.

Reply:
Ok, I know its bit of nit-picking but perhaps we can separate the code for setting the option to `News` and do it independently at the end instead of `else` . So basically at the end, if you don't have category, you just default it to "News"

Reply:
Thank you for the feedback as always @sdmcraft  . 

Now if we put that code independently at the end, as shown below - 

<img width="1040" alt="image" src="https://github.com/hlxsites/sunstar/assets/37831850/5c81f3b6-4d91-4bcb-98c0-bf2218ca657d">

Its just that its fine when there is no category then it will go down & select category as News BY default .  But when we have any other category - let say - "Press Release" then the moment we click on Filter, it will go down & change the Category from "Press Release" to "News" which will be wrong since that article is a "Press Release" article - 

<img width="1034" alt="image" src="https://github.com/hlxsites/sunstar/assets/37831850/00ec97a8-4c7d-4ed6-ad97-c0ddbe3fff97">


The moment we click on Filter, then it will change "Press Release" to "News" 

<img width="1065" alt="image" src="https://github.com/hlxsites/sunstar/assets/37831850/2ba89fe6-e090-46a6-a37c-4d72dfd22a04">

Hence that else is required. Please advise.